### DISCUSSION TITLE
Rotor simulation with Reynolds number effects at variable RPM

### DISCUSSION THREAD

**Discussion by MarcoPic99** [byuflowlab/FLOWUnsteady#136] [Q&A] [answered]
Hi!

I was reading your eVTOL example, where you simulate a maneuver and i noticed that, for all the propellers, you read just one polar at a fixed Reynolds number. This means that the aerodynamics coefficients of the propellers are assumed constant during the maneuver, right? So, if i would take into account the aero coefficients variation i need to run xfoil or provide a different polar at each time step ? Or there is a different way to do that?

Thank you in advance!

**Comment by EdoAlvarezR** [comment, marked as answer]
Hi Marco! That is correct. The Reynolds number is assumed to be constant throughout the maneuver. 

Currently, there isn't a clean way of providing different polars for different time steps, but it can certainly be done if you are brave enough to manually modify [the polars](https://github.com/byuflowlab/FLOWVLM/blob/master/src/FLOWVLM_rotor.jl#L81-L84) at each time step inside a custom-defined runtime function.

**Comment by MarcoPic99** [comment]
Thank you very much !

Marco

**Comment by EdoAlvarezR** [comment]
Related to #114

**Reply by MarcoPic99** [reply]
Hi ! I tried to implement this feature and seems to work. In the figure, i tested a small drone propeller during a maneuver and the Reynolds effect seems to be taken into account in the CT evaluation.
I think this feature could be useful for the simulation of small-scaled propellers which operates at low-Re regimes, where the effect on the aerodynamic coefficients could be not so neglegible.


![mod](https://github.com/byuflowlab/FLOWUnsteady/assets/154614588/7638ada5-76c3-4fde-bdd0-ba7cfd28ac04)

Maybe after i finish to test the function i wrote, i could upload the code here, if someone is interested.

Thank you

Marco

**Reply by EdoAlvarezR** [reply]
It would be great if you could share the code here!

**Reply by MarcoPic99** [reply]
Hi! sorry for the late response. Here is the function. I'm not a programmer and i'm not experienced in Julia, so I hope someone could improve it!  This function is then passed to "run_simulation" as ad additional input. I tested the function on several maneuvers and it works. Let me now if someone finds mistakes or have suggestions to improve it!

I hope this can be helpful to someone!


`function polaRe(sim, T)

 	rotor = sim.vehicle.rotor_systems[1][1] # Rotor 1
	
	# spline chord
        spline_k=5
        spline_s=0.001
        splines_s=nothing
        spline_bc="extrapolate"
        _spl_chord = uns.Dierckx.Spline1D(rotor.r, rotor.chord;
                                        k= spline_k,
                                        s=splines_s!=nothing ? splines_s[1] : spline_s, bc=spline_bc)
        spl_chord(x) = uns.Dierckx.evaluate(_spl_chord, x)

	alpha_vec = collect(-20:1:20) # Polar alpha range
	Re_vec=LinRange(257, 51252, 1991) # Polar Reynolds range
	
	# Read polar map from .csv file (it can be done outside the function)
	data_CL = Matrix(uns.CSV.read("polar3/CLmap.csv", uns.DataFrames.DataFrame, header=false))
	data_CD = Matrix(uns.CSV.read("polar3/CDmap.csv", uns.DataFrames.DataFrame, header=false)) 
	data_Cm = Matrix(uns.CSV.read("polar3/Cmmap.csv", uns.DataFrames.DataFrame, header=false)) 
	
	# Compute rotational velocity of the rotor
	omega = uns.get_RPM(sim.maneuver,1,T)*sim.RPMref

	
	airfoils = []
	for i in 1:size(sim.vehicle.rotor_systems[1][1].airfoils,1)

		# Compute Reynolds number for each radial station in airfoil .csv file
		pos = sim.vehicle.rotor_systems[1][1].airfoils[i][1] 
		x_airfoil = sim.vehicle.rotor_systems[1][1].airfoils[i][2].x
		y_airfoil = sim.vehicle.rotor_systems[1][1].airfoils[i][2].y
		roR = (rotor.hubR + pos*(rotor.rotorR-rotor.hubR))/rotor.rotorR
		ReD = (2*rotor.rotorR)/(1.54e-5)*omega/60*sqrt((2*pi*roR*rotor.rotorR)^2+(0.001*2*rotor.rotorR)^2)
		Re_c = ceil(Int, ReD*spl_chord(roR*rotor.rotorR)/(2*rotor.rotorR))
		
		#Interpolate the aerodynamic coefficients
		interp_CL = Interpolations.interpolate((alpha_vec, Re_vec), data_CL, Gridded(Linear()))
		interp_CD = Interpolations.interpolate((alpha_vec, Re_vec), data_CD, Gridded(Linear()))
		interp_Cm = Interpolations.interpolate((alpha_vec, Re_vec), data_Cm, Gridded(Linear()))

		CL_i = interp_CL(alpha_vec,Re_c)
		CD_i = interp_CD(alpha_vec,Re_c)
		Cm_i = interp_Cm(alpha_vec,Re_c)

		#Create new polar
		polar = vlm.ap.Polar(-1, alpha_vec, CL_i, CD_i, Cm_i; x=x_airfoil,y=y_airfoil)
		push!(airfoils, (pos, polar))
	
	end
	
	#update rotor with new polars
	rotor.airfoils = airfoils
	vlm._calc_airfoils(rotor, rotor.m, 1/5, false, []; rediscretize=false,
                        rfl_n_lower=15, rfl_n_upper=15,
                        rfl_r=14.0, rfl_central=true)

	sim.vehicle.rotor_systems[1][1] = rotor
	sim.vehicle.system.wings[1] = rotor
	sim.vehicle.wake_system.wings[1] = rotor

	return sim
   end`

**Reply by EdoAlvarezR** [reply]
Awesome, thanks for sharing! =]
